tests/unit/test_runner_metrics.py

```diff
@@ -90,6 +113,132 @@ def _create_runner_files(
+        str(runner_metrics_data.installed_timestamp),
+    )
+    shared_fs_mock.list_all.return_value = [runner_fs]
+
+    return runner_fs.runner_name
+
+
+def _setup_ghapi_mock(runner_names: set[str]) -> MagicMock:
+    """Setup a mocked GhApi object to return jobs for the given runners.
+
+    Args:
+        runner_names: The names of the runners to return jobs for.
+    """
+    ghapi_mock = MagicMock()
+    ghapi_mock.actions = MagicMock()
+    ghapi_mock.actions.list_jobs_for_workflow_run.return_value = {
+        "jobs": [
+            {
+                "created_at": TEST_JOB_CREATED_AT,
+                "started_at": TEST_JOB_STARTED_AT,
+                "runner_name": runner_name,
+            }
+            for runner_name in runner_names
+        ]
+    }
+    return ghapi_mock
+
+
+def _setup_ghapi_mock_multiple_pages(runner_name: Optional[str] = None) -> MagicMock:
+    """Setup a mocked GhApi object to return multiple pages for a workflow run
+
+    Args:
+        runner_name: If passed, make sure the run contains a job with that runner name.
+    """
+    no_of_pages = random.choice(range(1, 5))
+    no_of_jobs_per_page = random.choice(range(1, 4))
+    runner_names = [secrets.token_hex(16) for _ in range(no_of_pages * no_of_jobs_per_page)]
+
+    if runner_name:
+        runner_names[random.choice(range(no_of_pages))] = runner_name
+
+    ghapi_mock = MagicMock()
+    ghapi_mock.actions = MagicMock()
+
+    ghapi_mock.actions.list_jobs_for_workflow_run.side_effect = [
+        {
+            "jobs": [
+                {
+                    "created_at": TEST_JOB_CREATED_AT,
+                    "started_at": TEST_JOB_STARTED_AT,
+                    "runner_name": runner_names[i * no_of_jobs_per_page + j],
+                }
+                for j in range(no_of_jobs_per_page)
+            ]
+        }
+        for i in range(no_of_pages)
+    ] + [{"jobs": []}]
+
+    return ghapi_mock
+
+
+def _assert_runner_events_issued(
+    flavor: str,
+    issue_event_mock: MagicMock,
+    runner_metrics_data: RunnerMetrics,
+    stats: runner_metrics.IssuedMetricEventsStats,
+    queue_duration: Optional[float],
+):
+    """Assert that the runner events are issued.
+
+    Args:
+        flavor: The flavor of the runner.
+        issue_event_mock: The mock of the issue_event function.
+        runner_metrics_data: The runner metrics data.
+        stats: The stats returned by extract.
+        queue_duration: The queue duration in seconds.
+
+    """
+    assert stats == {
+        RunnerStart: 1,
+        RunnerStop: 1,
+    }
+    issue_event_mock.assert_any_call(
+        RunnerStart(
+            timestamp=runner_metrics_data.pre_job.timestamp,
+            flavor=flavor,
+            workflow=runner_metrics_data.pre_job.workflow,
+            repo=runner_metrics_data.pre_job.repository,
+            github_event=runner_metrics_data.pre_job.event,
+            # Ignore line break before binary operator
```

Comment:
This one can probably go into the configuration file since I think this is something where pylint disagrees with black